Casks/g/graalvm-jdk.rb

```diff
@@ -12,20 +12,11 @@
   homepage "https://www.graalvm.org/"
 
   livecheck do
-    url "https://www.oracle.com/java/technologies/downloads/"
-    regex(%r{/otn_software/java/jdk/(\d+(?:\.\d+)*)\+(\d+)/}i)
-    strategy :page_match do |page, regex|
-      major = page.scan(%r{href=.*?/technologies/javase-jdk(\d+)-doc-downloads\.html}i)
-                  .max_by { |match| Version.new(match[0]) }
-                  &.first
-      next if major.blank?
-
-      download_page = Homebrew::Livecheck::Strategy.page_content(
-        "https://www.oracle.com/java/technologies/javase-jdk#{major}-doc-downloads.html",
-      )
-      next if (download_page_content = download_page[:content]).blank?
-
-      download_page_content.scan(regex).map { |match| "#{match[0]},#{match[1]}" }
+    url "https://java.oraclecloud.com/currentJavaReleases?sortBy=releaseVersion"
+    regex(/jdk-(\d+(?:\.\d+)*)(?:-\d+)?\+(\d+)/i)
+    strategy :json do |json, regex|
+      match = json["items"][0]["releaseFullVersion"].match(regex)
+      "#{match[1]},#{match[2]}"
```

Comment:
```suggestion
    regex(/(?:jdk[._-])?(\d+(?:\.\d+)*)(?:-\d+)?\+(\d+)/i)
    strategy :json do |json, regex|
      json["items"]&.map do |item|
        match = item["releaseFullVersion"]&.match(regex)
        next if match.blank?

        "#{match[1]},#{match[2]}"
      end
```

A few things:

* We should make the `jdk-` prefix optional in the regex, so it will continue to match if they drop it from the `releaseFullVersion` text in the future.
* livecheck sorts matched versions and uses the highest one, so there's no need to artificially restrict this to the first item. The items are currently sorted from the highest version to lowest but that approach wouldn't work as expected if that ever changes in the future (something that we've experienced in other checks over the years). It's not required to only use the first item for this to work, so we can simply work with all the items instead (as we do in other checks).
* There's no guarantee that a field will exist in the JSON data, so we have to treat values like `json["items"]` as nilable and handle that appropriately.